### DISCUSSION TITLE
Database entries in invoices_meta

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#3423] [General]
Not sure if this is the right place but I will give it a try: 

I created a meta field for invoices (called "partial payment amount"). When I create a new invoice without filling in an entry in the meta field, it is not registered in the database in the db-table invoices_meta (expected behaviour). When I add an entry (amount of partial payment), a row for this invoice with the amount of the partial payment is added in this db-table (expected behaviour). 

However, if I simply mark an invoice as paid, an entry is nevertheless added for this invoice in the db-table invoices_meta whereas the value is set to "NULL". So, ideally all invoices will be added to the db-table invoices_meta eventually (which means they are paid) but in my view, it is unnecessary data. 

Is there a way to avoid an entry "NULL" in the db-table invoices_meta simply because they are marked as paid and maybe additionlly, a way to automatically delete entries if there value is "NULL" again (so that the table invoice_meta includes only rows with a value not NULL)?

**Comment by kevinpapst** [comment]
I will have to check that, the meta-field row should be created always. 
Seems there is something missing when creating a new invoice. 
Did you wanted to share your findings or do you really care about these empty rows?

**Comment by BeckeBauer** [comment]
I found it be chance when I tried to understand how invoices numbers are created (see discussion #3427) and changed the invoice IDs manually - however, since the invoice ID has no effect on the counter of an individual customer, we do not need to further discuss this matter...
